core/src/oracle_state.rs

```diff
@@ -276,6 +279,12 @@ impl<'a> OraclePool<'a> {
         )
         .expect("Failed to parse scanIDs.json");
 
+        let wallet_height = get_wallet_status()?.height;
+        let block_height = current_block_height()?;
+        if wallet_height != block_height {
+            warn!("Set-scans may not have completed yet, scan height: {}, current blockchain height: {}", wallet_height, block_height);
```

Comment:
I wonder if logging a warning is not enough here. If the scan is not completed yet, we can't be sure that our code (main loop, etc.) will work as intended. Maybe we should wait for the wallet scan to complete before going further. What do you think?

Reply:
Good question! I don't think waiting here is a good idea since this ctor is called from multiple places. Maybe extracting scansIDs.json creation/loading and wallet scan registration somewhere so that it could be called once on oracle start will give us a better place to wait for the wallet scanning to finish.